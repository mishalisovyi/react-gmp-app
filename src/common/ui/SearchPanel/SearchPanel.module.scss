.SearchPanel {
  &__controls {
    display: grid;
    grid-template-columns: 7fr 2fr;
    column-gap: 14px;
  }
}
