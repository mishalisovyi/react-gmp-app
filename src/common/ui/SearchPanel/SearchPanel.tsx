import React, { memo, useMemo, useRef } from 'react';
import { ButtonType } from 'common/enums';
import { Button, TextField } from 'common/ui';
import styles from './SearchPanel.module.scss';

interface SearchPanelProps {
  labelText?: string;
  value?: string;
  onSearch: (searchTerm: string) => void;
}

function SearchPanelComponent({ labelText, value, onSearch }: SearchPanelProps) {
  const textFieldRef = useRef<HTMLInputElement>(null);

  const handleSearchButtonClick = () => {
    const inputElement = textFieldRef.current as HTMLInputElement;

    onSearch(inputElement.value);
  };

  const label = useMemo(() => {
    return labelText ? <h1 className="typography-heading">{labelText}</h1> : null;
  }, [labelText]);

  return (
    <>
      {label}
      <div className={styles['SearchPanel__controls']}>
        <TextField ref={textFieldRef} value={value} placeholder="What do you want to watch?" />
        <Button type={ButtonType.Primary} text="Search" onClick={handleSearchButtonClick} />
      </div>
    </>
  );
}

SearchPanelComponent.defaultProps = {
  labelText: '',
  value: '',
};

export const SearchPanel = memo(SearchPanelComponent);
