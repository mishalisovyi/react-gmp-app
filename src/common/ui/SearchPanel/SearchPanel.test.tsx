import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { SearchPanel } from './SearchPanel';

const handleSearch = jest.fn();

it('should render search panel', () => {
  const tree = renderer.create(<SearchPanel labelText="test label" value="test value" onSearch={handleSearch} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should handle search', async () => {
  const { container } = render(<SearchPanel labelText="test label" value="test value" onSearch={handleSearch} />);

  const [inputElement] = container.getElementsByTagName('input');

  const searchTerm = 'test string';

  await userEvent.clear(inputElement);
  await userEvent.type(inputElement, searchTerm);
  await userEvent.click(screen.getByText('Search'));

  expect(handleSearch).toHaveBeenCalledWith(searchTerm);
});
