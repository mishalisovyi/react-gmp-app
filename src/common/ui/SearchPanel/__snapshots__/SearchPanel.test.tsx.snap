// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render search panel 1`] = `
Array [
  <h1
    className="typography-heading"
  >
    test label
  </h1>,
  <div>
    <input
      defaultValue="test value"
      placeholder="What do you want to watch?"
      type="text"
    />
    <button
      className="button button--primary"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </div>,
]
`;
